src/main/java/seedu/address/model/person/Name.java

```diff
@@ -10,13 +10,15 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain alphanumeric characters and spaces, "
+                    + "and its length should be between 1 and 30 (including spaces)";
 
+    public static final String NAME_CHARACTER_LIMIT = "{0,29}";
```

Comment:
30 might be too restrictive. There are many names out there that is more than 30 characters. The longest name in the world is also close to 800 characters. To be inclusive of everyone, perhaps you could consider 800 characters as the limit?